File: Binary Tree/Binary Search Tree/bst.js
class node{
    constructor(element){
        this.element = element;
        this.element.left = null;
        this.element.right = null;
    }
}

class bst{
    constructor(){
        this.root = null;
    }
    createBst(root, element) {
        var t = root;
        if (root.data === 0){
            var newnode  = new node(element);
            root = newnode;
            return root;
        }
        else{
            while(t){
                if (element == t){
                    return root;
                }
                else if (element < t){
                    t = t.left;
                }
                else{
                    t = t.right;
                }
            }
            var newnode = new node (element);
            if (element<t){
                t.left = newnode;
            }
            else{
                t.right = newnode;
            }
            return root;
        }
    }

    inOrder(root){
        if (root){
            this.inOrder(root.left);
            console.log(root.data);
            this.inOrder(root.right);
        }
    }
};



const tree = new bst();
var root = new node(0);
root = tree.createBst(root, 40);
root = tree.createBst(root, 20);
root = tree.createBst(root, 30);
root = tree.createBst(root, 60);
root = tree.createBst(root, 50);
root = tree.createBst(root, 10);
root = tree.createBst(root, 70);
tree.inOrder(root);

File: Graph/bst.cpp
#include<bits/stdc++.h>
using namespace std;

struct Node{
    int data;
    Node *left;
    Node *right;
};

Node *addNode(Node *head, int data){
    Node *newnode = new Node();
    newnode->data = data;
    newnode->left = NULL;
    newnode->right = NULL;
    if(!head){
        return newnode;
    }
    Node *temp = head;
    while(1){
        if(data < temp->data){
            if(temp->left){
                temp = temp->left;
            }
            else{
                temp->left = newnode;
                break;
            }
        }
        else{
            if(temp->right){
                temp = temp->right;
            }
            else{
                temp->right = newnode;
                break;
            }
        }
    }
    return head;
}

void preOrder(Node *head){
    if(head){
        preOrder(head->left);
        cout<<head->data<<" ";
        preOrder(head->right);
    }
}

int main(){
    Node *head = NULL;
    head = addNode(head, 40);
    head = addNode(head, 20);
    head = addNode(head, 60);
    head = addNode(head, 10);
    head = addNode(head, 70);
    head = addNode(head, 50);
    preOrder(head);
    return 0;
}